core/src/test/scala/org/http4s/TestInstances.scala

```diff
@@ -191,6 +191,27 @@ trait TestInstances {
         arbitrary[Header.Raw]
       )
     }
+
+  implicit lazy val arbitraryServerSentEvent: Arbitrary[ServerSentEvent] = {
+    def singleLineString: Gen[String] =
+      arbitrary[String] suchThat { s => !s.contains("\r") && !s.contains("\n") }
+    Arbitrary(for {
+      data <- singleLineString
```

Comment:
A data field can represent newlines.  I don't know how to represent carriage returns without an ad hoc escaping mechanism.